### PR TITLE
Rm retriever kwargs

### PR DESCRIPTION
**Pull request by baskaryan**
Doesn't actually limit the Retriever interface but hopefully in practice it does

### CODE REVIEW

File: langchain/schema.py [hunk 1]
```diff
@@ -370,7 +370,40 @@ class Document(Serializable):
 
 
 class BaseRetriever(ABC):
-    """Base interface for a retriever."""
+    """Minimal interface for a Retriever."""
+
+    @abstractmethod
+    def get_relevant_documents(
+        self, query: str, *, callbacks: Callbacks = None, **kwargs: Any
```
Comment (baskaryan) [inline langchain/schema.py:377]: need to think a bit more about backwards compatibility here

File: langchain/retrievers/self_query/base.py [hunk 2]
```diff
@@ -121,11 +117,7 @@ def _get_relevant_documents(
         return docs
 
     async def _aget_relevant_documents(
-        self,
-        query: str,
-        *,
-        run_manager: Optional[AsyncCallbackManagerForRetrieverRun],
-        **kwargs: Any,
+        self, query: str, *, run_manager: Optional[AsyncCallbackManagerForRetrieverRun]
```
Comment (baskaryan) [inline langchain/retrievers/self_query/base.py:120]: just noticing there's decent amount of classes where run_manager is typed as Optional, some where it has a default val of None. on base it's not optional and has no default.  guessing discrepancy is unintentional @hinthornw @nfcampos?

Comment (baskaryan) [reply langchain/retrievers/self_query/base.py:120]: will save updating that for a sep pr

Comment (hinthornw) [reply langchain/retrievers/self_query/base.py:120]: It should be required. I had originally incorrectly made optional

**Pull request merged**